libs/blocks/aside/aside.js

```diff
@@ -24,23 +28,59 @@ function decorateLayout(el) {
   return foreground;
 }
 
-function decorateContent(el, isInline) {
+function decorateContent(el, type, size) {
   if (!el) return;
-  const text = el.querySelector('h1, h2, h3, h4, h5, h6')?.closest('div');
+  const text = el.querySelector('h1, h2, h3, h4, h5, h6, a')?.closest('div');
+  const picture = text?.querySelector('picture');
+  const iconArea = picture ? (picture.closest('p') || createTag('p', null, picture)) : null;
   text?.classList.add('text');
+  if (text && !text.querySelector('p')) {
+    const buttons = text.querySelectorAll('em a, strong a');
+    let btnWrap = null;
+    if (buttons[0] && !buttons[0].closest('p')) {
+      btnWrap = document.createElement('p');
+      buttons.forEach((button) => btnWrap.append(button.parentElement));
+    }
+    const desc = createTag('p', null, text.innerHTML);
+    text.innerHTML = '';
+    if (iconArea) text.append(iconArea);
+    text.append(desc);
+    if (btnWrap) text.append(btnWrap);
+  }
+  iconArea?.classList.add('icon-area');
+  decorateButtons(el);
   const headings = text?.querySelectorAll('h1, h2, h3, h4, h5, h6');
   const heading = headings?.[headings.length - 1];
-  heading?.classList.add(`${isInline ? 'heading-S' : 'heading-XL'}`);
-  heading?.nextElementSibling?.classList.add(`${isInline ? 'body-M' : 'body-S'}`);
-  heading?.previousElementSibling?.classList.add('detail-M');
+  const isInline = type === 'inline';
+  const isNotification = type === 'notification';
+  let headingClass = 'heading-XL';
+  let bodyClass = 'body-S';
+  if (isNotification && (size === 'extra-small' || size === 'small')) {
+    headingClass = '';
+  }
+  if ((isNotification && (size === 'extra-small' || size === 'small' || size === 'large')) || isInline) {
+    bodyClass = 'body-M';
+  }
+  if ((isNotification && size === 'medium') || isInline) {
+    headingClass = 'heading-S';
+  }
+  if (isNotification && size === 'large') {
+    headingClass = 'heading-L';
+  }
+  heading?.classList.add(headingClass);
+  const bodyCopy = heading?.nextElementSibling.classList.length === 0 ? heading.nextElementSibling : text?.querySelector('p:not([class])');
+  bodyCopy?.classList.add(bodyClass);
+  const body = createTag('div', { class: 'body-area' });
+  bodyCopy.insertAdjacentElement('beforebegin', body);
+  body.append(bodyCopy);
+  const prevClasses = heading?.previousElementSibling?.classList;
+  if (prevClasses?.length === 0) prevClasses.add('detail-M');
   el.querySelector(':scope > div:not([class])')?.classList.add('image');
-  const image = text?.querySelector('img');
-  if (image) image.closest('p')?.classList.add('icon-area');
 }
 
 export default function init(el) {
   const foreground = decorateLayout(el);
-  const isInline = el.className.includes('inline');
-  decorateContent(foreground, isInline);
-  decorateButtons(foreground);
+  const type = asideTypes?.find((asideType) => el?.className?.indexOf(asideType) !== -1);
```

Comment:
I'd remove most of these optionals.

I'm seeing a lot of excessive use of optionals in a couple other PRs, too.

There's no need to do an optional for every object. We know an el will always be passed in. We know the element API will always have classList / className even if they're empty.

Please try to look through this file and see what is really needed as optional and what isn't.

Reply:
I removed the ones in `init`. Like you said, all of those we unnecessary. The ones in `decorateContent` are there to deal with variations in authoring. Do you want me to convert them to if statements? In northstar, we use optional chaining to help maintain a flat structure so that's what I'm used to. Do you prefer nested ifs to optional chaining?